core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-or-update-solution-modal.component.ts

```diff
@@ -91,8 +96,11 @@ export class AddOrUpdateSolutionModalComponent
   }
 
   shouldAdditionalSubmitButtonBeShown(): boolean {
-    let interactionSpecs = INTERACTION_SPECS[
-      this.stateInteractionIdService.savedMemento];
+    const InteractionSpec:
+      {[key: string]: {show_generic_submit_button: boolean} } = (
+        INTERACTION_SPECS);
```

Comment:
I think we already have a type for `INTERACTION_SPECS`, can you check the codebase?

Reply:
Yes, we have.Updated